File: src/main/golang/leetcode/editor/cn/[215]Kth Largest Element in an Array.go
package lc

//leetcode submit region begin(Prohibit modification and deletion)
func findKthLargest(nums []int, k int) int {
	n := len(nums)
	var ans int
	var quickSort func(l, r int)
	quickSort = func(l, r int) {
		if l > r {
			return
		}
		pivot := nums[l]
		nums[l], nums[r] = nums[r], nums[l]
		var p0 = l
		for p1 := l; p1 < r; p1++ {
			if nums[p1] < pivot {
				nums[p0], nums[p1] = nums[p1], nums[p0]
				p0++
			}
		}

		nums[p0], nums[r] = nums[r], nums[p0]
		if n-p0 == k {
			ans = nums[p0]
			return
		} else if n-p0 < k {
			quickSort(l, p0-1)
		} else {
			quickSort(p0+1, r)
		}
	}
	quickSort(0, n-1)
	return ans
}

//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[215]Kth Largest Element in an Array.py
from typing import List


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:

        n = len(nums)

        def quickSort(nums, l, r):
            if l >= r:
                return nums[l]
            i, j = l, r
            while i < j:
                while i < j and nums[j] >= nums[l]:
                    j -= 1
                while i < j and nums[i] <= nums[l]:
                    i += 1
                nums[i], nums[j] = nums[j], nums[i]
            nums[i], nums[l] = nums[l], nums[i]
            if i == n - k:
                return nums[i]
            elif i < n - k:
                return quickSort(nums, i + 1, r)
            else:
                return quickSort(nums, l, i - 1)

        return quickSort(nums, 0, len(nums) - 1)
# leetcode submit region end(Prohibit modification and deletion)
